### PR TITLE
feat: support #:auto-value for defparam and defthing

### PR DESCRIPTION
**Pull request by sorawee**
Manually specifying `#:value` could be cumbersome. An example is `html-empty-tags` from the `xml` collection, which is a list of more than 10 symbols.
It is also error-prone, and could be out-of-sync from the actual value.

This PR adds a support for `#:auto-value` in `defparam` and `defthing` to automatically query the value from the `for-label` binding.

Thanks to @LiberalArtist for the help in https://github.com/racket/racket/pull/4807.

### CODE REVIEW

File: scribble-lib/scribble/private/manual-proc.rkt [hunk 1]
```diff
@@ -140,6 +141,28 @@
...
                        #:defaults ([value #'no-value]))))
+
+ (define-splicing-syntax-class (autoable-value-kw name proc)
+   #:description "#:value or #:auto-value keyword"
+   (pattern (~seq #:value extracted))
+   (pattern (~seq #:auto-value)
+            #:do [(define binding-info (identifier-binding name #f))
+                  (unless (list? binding-info)
+                    (raise-syntax-error #f "not imported for-label" name))
+                  ;; the third result is nominal-from-mod, which is what we want
```
Comment (sorawee) [inline scribble-lib/scribble/private/manual-proc.rkt:152]: I'm actually not sure about this...

**Comment by sorawee** [comment]
This strategy is not foolproof though. For example, https://github.com/racket/scribble/blob/master/scribble-doc/scribblings/scribble/utils.rkt#L12-L13 bounces the for-label binding, but not the phase 0 binding. So without `(require (for-label racket/base))`, `current-readtable` would be identified as originated from `utils.rkt`, which is not the case. I'm not sure if there's a way to fix the problem. We could also just document the limitation and call it a day...

**Comment by sorawee** [comment]
Switching from the third result to the first result (`nominal-from-mod` -> `from-mod`) seems to help with the above issue.

**Comment by sorawee** [comment]
Addressed feedback by @rocketnia.

File: scribble-doc/scribblings/scribble/manual.scrbl [hunk 2]
```diff
@@ -1379,6 +1392,11 @@ If @racket[#:value value-expr-datum] is given, @racket[value-expr-datum]
 is typeset using @racket[racketblock0] and included in the documentation.
 Wide values are put on a separate line.
 
+@racket[#:auto-value] is similar to @racket[#:value value-expr-datum],
```
Comment (mflatt) [inline scribble-doc/scribblings/scribble/manual.scrbl:1395]: Nit: I'd rephrase to avoid starting the sentence with a lowercase keyword, maybe "Supplying #:auto-value ...".

File: scribble-test/tests/scribble/docs/manual-ex.rkt [hunk 3]
```diff
@@ -20,3 +20,14 @@
 (struct pn (x y))
 
 (define v 10)
+
+(require racket/fixnum
```
Comment (mflatt) [inline scribble-test/tests/scribble/docs/manual-ex.rkt:24]: Nit: I'd prefer to keep `require`s at the top.

**Review by mflatt — APPROVED** [review]
Looks ok to me. Ignore nit-picking comments as you see fit.

**Comment by sorawee** [comment]
Oof, I should have bumped the version and added a history note. Will push a fix.
